corehq/apps/userreports/pillow.py

```diff
@@ -53,6 +54,23 @@
 LONG_UCR_LOGGING_THRESHOLD = 0.5
 
 
+class WarmShutdown(object):
+    # modified from https://stackoverflow.com/a/50174144
+
+    shutting_down = False
+
+    def __enter__(self):
+        self.current_handler = signal.signal(signal.SIGTERM, self.handler)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self.shutting_down:
+            exit(0)
```

Comment:
shouldn't this only exist with `0` if there are no exceptions?

Reply:
Since the code does not catch any exceptions, won't those bubble up and this line only hit if it got here without exceptions? Or does `with` catch those automatically.

Reply:
the __exit__ method will still get executed in the case of an exception happening and since it's not handled here it will get swallowed:
```python
# exit_test.py
import sys

class A(object):
    def __enter__(self):
        print('in A')

    def __exit__(self, exc_type, exc_value, traceback):
        print('exiting A', exc_type)
        sys.exit(0)


with A():
    raise Exception('asdf')
```
```
$ python exit_test.py
in A
exiting A <class 'Exception'>

$ echo $?
0

```

Reply:
Cool, had to reread the docs on context managers, but should be fixed now